nemo/core/neural_types/elements.py

```diff
@@ -69,28 +70,35 @@ class ElementType(ABC):
     We are relying on Python for inheritance checking"""
 
     def __str__(self):
+        if torch.jit.is_scripting():
+            return "SuppressedForTorchScript"
         return self.__doc__
 
     def __repr__(self):
+        if torch.jit.is_scripting():
+            return "SuppressedForTorchScript"
         return self.__class__.__name__
 
     @property
-    def type_parameters(self) -> Dict:
+    def type_parameters(self) -> Dict[str, Any]:
         """Override this property to parametrize your type. For example, you can specify 'storage' type such as
         float, int, bool with 'dtype' keyword. Another example, is if you want to represent a signal with a
         particular property (say, sample frequency), then you can put sample_freq->value in there.
         When two types are compared their type_parameters must match."""
         return {}
 
     @property
-    def fields(self) -> Optional[Tuple]:
+    def fields(self):
         """This should be used to logically represent tuples/structures. For example, if you want to represent a
         bounding box (x, y, width, height) you can put a tuple with names ('x', y', 'w', 'h') in here.
         Under the hood this should be converted to the last tesnor dimension of fixed size = len(fields).
         When two types are compared their fields must match."""
         return None
 
     def compare(self, second) -> NeuralTypeComparisonResult:
+        if torch.jit.is_scripting():
+            # Suppress in torch.jit.script
+            return NeuralTypeComparisonResult.SAME
```

Comment:
Why should this be hard coded ? It will give incorrect result for any scripted model ?

Reply:
Thanks, removed (now suppression here is redundant, this was an artifact from the first attempt)